.tekton/tasks/check-pr-label.yaml

```diff
@@ -0,0 +1,70 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: pr-labels
+spec:
+  description: |
+    check the pull request label for which the test is running on staging is ready to merge or not.
+  params:
+    - name: match_label
+    - name: pull_request_number
+    - name: TARGET_GH_OWNER
+      description: Owner of GitHub repository of the infra-deployments code
+      default: redhat-appstudio
+    - name: TARGET_GH_REPO
+      description: GitHub repository of the infra-deployments code
+      default: infra-deployments
+  results:
+    - name: matched-merge-label
+  workspaces:
+    - name: shared-file-path
+  steps:
+    - name: check-pr-label
+      image: quay.io/redhat-appstudio/github-app-token@sha256:b4f2af12e9beea68055995ccdbdb86cfe1be97688c618117e5da2243dc1da18e
+      workingDir: $(workspaces.shared-file-path.path)
+      env:
+        - name: GITHUB_API_URL
+          value: https://api.github.com
+        - name: PULL_REQUEST_NUMBER
+          value: $(params.pull_request_number)
+        - name: TARGET_GH_OWNER
+          value: "$(params.TARGET_GH_OWNER)"
+        - name: TARGET_GH_REPO
+          value: "$(params.TARGET_GH_REPO)"
+      script: |
+        #!/usr/bin/env python3
```

Comment:
I would +1 what @Roming22 and @enarha have noted wrt this.
I would also add the possible future motiviation that if we create script files vs. in-lining, it could help facilitate reuse both as 
1) other components of RHTAP start to employ the gating to staging/prod that @prietyc123 is leading here
2) and even more futuristic, and if it has been mentioned elsewhere, I missed it, but once we are able to reuse this stuff a few  times, we may get it clean enough wrt abstractions and parameterization that we could start providing this as part of the RHTAP bundles that customers ultimately use (either internally in RH or even project dance) .... this type of test env to prod CI/CD flow are of course the norm, vs. unique to us here at RH

Reply:
Done